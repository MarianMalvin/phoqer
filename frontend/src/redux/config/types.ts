const generalTypes = {
    // TOGGLE_CHAT_DRAWER
    TOGGLE_CHAT_DRAWER: 'TOGGLE_CHAT_DRAWER',
    // OFFERS_SEARCH_LOCAL_PARAMS
    OFFERS_SEARCH_LOCAL_PARAMS: 'OFFERS_SEARCH_LOCAL_PARAMS',
};

export default generalTypes;
