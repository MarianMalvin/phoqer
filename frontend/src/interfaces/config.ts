import { ISearch } from './general';

export interface IStateConfig {
    chatDrawer: boolean;
    searchParams: ISearch;
}
