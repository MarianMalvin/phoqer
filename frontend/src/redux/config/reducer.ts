import { combineReducers } from 'redux';

import chatDrawer from './chatDrawer/reducer';
import searchParams from './searchParams/reducer';

const config = combineReducers({
    chatDrawer,
    searchParams,
});

export default config;
