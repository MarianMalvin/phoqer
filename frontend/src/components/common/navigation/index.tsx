import React, { ReactElement } from 'react';

import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import { useRouter } from 'next/router';
import { createUseStyles } from 'react-jss';

import useTrans from '../../../hooks/trans.hook';
import { ITabs } from '../../../interfaces';
import { Theme } from '../../../utils/theming/theme';
import Badge from '../badge';

const useStyles = createUseStyles((theme: Theme) => ({
    item: {
        margin: theme.rem(0.2, 0),

        ...theme.media(768).max({
            margin: theme.rem(1, 0),
        }),
    },
    button: {
        display: 'flex',
        alignItems: 'center',
        width: '100%',
        padding: theme.rem(1.4, 1),
        borderRadius: theme.radius,
        fontSize: theme.rem(1.4),
        transition: theme.transitions[0],
        color: theme.palette.black[0],

        ...theme.media(768).max({
            padding: theme.rem(1.4, 2),
            background: theme.palette.gray[0],
        }),

        ...theme.hover({
            background: theme.palette.gray[0],
            color: theme.palette.primary[0],
        }),
    },
    text: {
        margin: theme.rem(0, 1),
    },
}));

interface NavigationItemProp {
    tab: ITabs;
}

export const NavigationItem = ({ tab }: NavigationItemProp): ReactElement => {
    const css = useStyles();
    const trans = useTrans();
    const history = useRouter();

    const handleClick = (): void => {
        if (tab.onClick) {
            tab.onClick();
            return;
        }

        if (tab.link) history.push(tab.link);
    };

    return (
        <li className={css.item}>
            <button className={css.button} type="button" onClick={handleClick}>
                {tab.icon ? <FontAwesomeIcon icon={tab.icon} /> : null}
                <span className={css.text}>{trans(tab.text)}</span>
                {tab?.count ? <Badge>{tab.count}</Badge> : null}
            </button>
        </li>
    );
};

interface IProps {
    className?: string;
    tabs: ITabs[];
}

const Navigation = ({ className, tabs }: IProps): ReactElement => {
    return (
        <ul className={className}>
            {tabs.map<ReactElement>(item => (
                <NavigationItem key={item.id} tab={item} />
            ))}
        </ul>
    );
};

export default Navigation;
