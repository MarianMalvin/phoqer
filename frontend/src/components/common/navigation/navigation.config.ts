import { faCaretSquareLeft } from '@fortawesome/free-regular-svg-icons/faCaretSquareLeft';
import { faChartBar } from '@fortawesome/free-regular-svg-icons/faChartBar';
import { faCommentAlt } from '@fortawesome/free-regular-svg-icons/faCommentAlt';
import { faEnvelope } from '@fortawesome/free-regular-svg-icons/faEnvelope';
import { faPlusSquare } from '@fortawesome/free-regular-svg-icons/faPlusSquare';
import { faSlidersH } from '@fortawesome/free-solid-svg-icons/faSlidersH';

import { ITabs, ITabsNum } from '../../../interfaces';
import routes from '../../../utils/routes';

const temp: ITabsNum = {
    messages: 4,
    notifications: 5,
};

export const getBaseNavList = (count: ITabsNum = temp): ITabs[] => [
    {
        id: 'my-offers',
        text: 'my_offers',
        link: routes.my_offers(),
        icon: faCaretSquareLeft,
    },
    {
        id: 'chat',
        text: 'chat',
        link: routes.chat.item(),
        icon: faEnvelope,
        count: count?.messages,
    },
    {
        id: 'notifications',
        text: 'notifications',
        link: routes.notifications,
        icon: faCommentAlt,
        count: count?.notifications,
    },
    {
        id: 'referral',
        text: 'invite_friends',
        link: routes.referral,
        icon: faPlusSquare,
    },
    {
        id: 'settings',
        text: 'settings',
        link: routes.settings(),
        icon: faSlidersH,
    },
    {
        id: 'analytics',
        text: 'analytics',
        link: routes.analytics,
        icon: faChartBar,
    },
];
