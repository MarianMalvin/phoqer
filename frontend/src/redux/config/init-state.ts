import { IStateConfig } from '../../interfaces';

const configInit: IStateConfig = {
    drawer: false,
    chatDrawer: false,
    searchParams: {
        search: null,
        category: null,
        sub_category: null,
        period: null,
        status: null,
        ordering: null,
        max_price: null,
        min_price: null,
        top: null,
        no_deposit: null,
        is_deliverable: null,
    },
};

export default configInit;
