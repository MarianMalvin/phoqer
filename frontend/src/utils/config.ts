import { Themes } from '../interfaces';

const host = process.env.NODE_ENV === 'development' ? 'http://dev.phoqer.com' : '';
const api = host + '/api/';

const config = {
    baseUrl: {
        development: (version = 'v1'): string => api + version,
        test: (version = 'v1'): string => api + version,
        production: (version = 'v1'): string => (process.browser ? api + version : 'http://backend:8001'),
    },
    socketUrl: (version = 'v1'): string => 'ws://' + (api.split('://')[1] || 'phoqer.com/api/') + version,
    uploadsUrl: (version = 'v2'): string => `${api}${version}/upload`,
    host: (lang = 'pl'): string => `${host}/${lang}`,
    img: host,
    themes: ['white', 'black'] as Themes[],
    googleApiKey: process.env.NEXT_PUBLIC_GOOGLE_MAP_API_KEY,
    appSecretKey: process.env.NEXT_PUBLIC_SECRET_KEY,
};

export default config;
