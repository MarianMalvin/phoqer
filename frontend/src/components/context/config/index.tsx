import React, { createContext, ReactElement, useState } from 'react';

import Cookies from 'js-cookie';

import { IConfig } from '../../../interfaces';
import { addMonthToDate } from '../../../utils/helpers';
import notificationsModal from '../../common/modal/notifications-modal';

const init: IConfig = {
    warning: true,
    hideSearchFilters: false,
    hideTopSearchQuery: false,
    hideTopOffers: true,
    offerCardSize: 'big',
};
export type IConfigValue = [config: IConfig, setConfig: (value: IConfig) => void];
export const Config = createContext<IConfigValue>([init, () => undefined]);

interface IProps {
    value?: IConfig | null;
    children: JSX.Element | JSX.Element[] | ReactElement;
}

const ConfigProvider = ({ children, value }: IProps): ReactElement => {
    const [config, setConfig] = useState<IConfig>(value || init);
    const handleConfig = (data: IConfig): void => {
        try {
            Cookies.set('phoqer_config', JSON.stringify(data), { expires: addMonthToDate(1) });
            setConfig(data);
        } catch (error) {
            notificationsModal('error');
            setConfig(init);
        }
    };
    return <Config.Provider value={[config, handleConfig]}>{children}</Config.Provider>;
};

export default ConfigProvider;
