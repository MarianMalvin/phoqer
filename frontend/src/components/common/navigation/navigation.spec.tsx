import { render } from '../../../tests/tests.utils';

import { getBaseNavList } from './navigation.config';

import Navigation from './index';

describe('Navigation component', () => {
    it('matches snapshot', () => {
        const { asFragment } = render(<Navigation tabs={getBaseNavList()} />);
        expect(asFragment()).toMatchSnapshot();
    });

    it('matches snapshot', () => {
        const { asFragment } = render(<Navigation className="test" tabs={getBaseNavList()} />);
        expect(asFragment()).toMatchSnapshot();
    });
});
