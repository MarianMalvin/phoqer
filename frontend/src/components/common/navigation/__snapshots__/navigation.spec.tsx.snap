// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navigation component matches snapshot 1`] = `
<DocumentFragment>
  <ul>
    <li
      class="item-0-2-1"
    >
      <button
        class="button-0-2-2"
        type="button"
      >
        <svg
          aria-hidden="true"
          class="svg-inline--fa fa-caret-square-left fa-w-14 "
          data-icon="caret-square-left"
          data-prefix="far"
          focusable="false"
          role="img"
          viewBox="0 0 448 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M272 157.1v197.8c0 10.7-13 16.1-20.5 8.5l-98.3-98.9c-4.7-4.7-4.7-12.2 0-16.9l98.3-98.9c7.5-7.7 20.5-2.3 20.5 8.4zM448 80v352c0 26.5-21.5 48-48 48H48c-26.5 0-48-21.5-48-48V80c0-26.5 21.5-48 48-48h352c26.5 0 48 21.5 48 48zm-48 346V86c0-3.3-2.7-6-6-6H54c-3.3 0-6 2.7-6 6v340c0 3.3 2.7 6 6 6h340c3.3 0 6-2.7 6-6z"
            fill="currentColor"
          />
        </svg>
        <span
          class="text-0-2-5"
        >
          My offers
        </span>
      </button>
    </li>
    <li
      class="item-0-2-1"
    >
      <button
        class="button-0-2-2"
        type="button"
      >
        <svg
          aria-hidden="true"
          class="svg-inline--fa fa-envelope fa-w-16 "
          data-icon="envelope"
          data-prefix="far"
          focusable="false"
          role="img"
          viewBox="0 0 512 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M464 64H48C21.49 64 0 85.49 0 112v288c0 26.51 21.49 48 48 48h416c26.51 0 48-21.49 48-48V112c0-26.51-21.49-48-48-48zm0 48v40.805c-22.422 18.259-58.168 46.651-134.587 106.49-16.841 13.247-50.201 45.072-73.413 44.701-23.208.375-56.579-31.459-73.413-44.701C106.18 199.465 70.425 171.067 48 152.805V112h416zM48 400V214.398c22.914 18.251 55.409 43.862 104.938 82.646 21.857 17.205 60.134 55.186 103.062 54.955 42.717.231 80.509-37.199 103.053-54.947 49.528-38.783 82.032-64.401 104.947-82.653V400H48z"
            fill="currentColor"
          />
        </svg>
        <span
          class="text-0-2-5"
        >
          My messages
        </span>
        <span
          class="num-0-2-6"
        >
          4
        </span>
        <svg
          aria-hidden="true"
          class="svg-inline--fa fa-chevron-down fa-w-14 "
          data-icon="chevron-down"
          data-prefix="fas"
          focusable="false"
          role="img"
          viewBox="0 0 448 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M207.029 381.476L12.686 187.132c-9.373-9.373-9.373-24.569 0-33.941l22.667-22.667c9.357-9.357 24.522-9.375 33.901-.04L224 284.505l154.745-154.021c9.379-9.335 24.544-9.317 33.901.04l22.667 22.667c9.373 9.373 9.373 24.569 0 33.941L240.971 381.476c-9.373 9.372-24.569 9.372-33.942 0z"
            fill="currentColor"
          />
        </svg>
      </button>
    </li>
    <li
      class="item-0-2-1"
    >
      <button
        class="button-0-2-2"
        type="button"
      >
        <svg
          aria-hidden="true"
          class="svg-inline--fa fa-comment-alt fa-w-16 "
          data-icon="comment-alt"
          data-prefix="far"
          focusable="false"
          role="img"
          viewBox="0 0 512 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M448 0H64C28.7 0 0 28.7 0 64v288c0 35.3 28.7 64 64 64h96v84c0 7.1 5.8 12 12 12 2.4 0 4.9-.7 7.1-2.4L304 416h144c35.3 0 64-28.7 64-64V64c0-35.3-28.7-64-64-64zm16 352c0 8.8-7.2 16-16 16H288l-12.8 9.6L208 428v-60H64c-8.8 0-16-7.2-16-16V64c0-8.8 7.2-16 16-16h384c8.8 0 16 7.2 16 16v288z"
            fill="currentColor"
          />
        </svg>
        <span
          class="text-0-2-5"
        >
          Notifications
        </span>
        <span
          class="num-0-2-6"
        >
          5
        </span>
      </button>
    </li>
    <li
      class="item-0-2-1"
    >
      <button
        class="button-0-2-2"
        type="button"
      >
        <svg
          aria-hidden="true"
          class="svg-inline--fa fa-sliders-h fa-w-16 "
          data-icon="sliders-h"
          data-prefix="fas"
          focusable="false"
          role="img"
          viewBox="0 0 512 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M496 384H160v-16c0-8.8-7.2-16-16-16h-32c-8.8 0-16 7.2-16 16v16H16c-8.8 0-16 7.2-16 16v32c0 8.8 7.2 16 16 16h80v16c0 8.8 7.2 16 16 16h32c8.8 0 16-7.2 16-16v-16h336c8.8 0 16-7.2 16-16v-32c0-8.8-7.2-16-16-16zm0-160h-80v-16c0-8.8-7.2-16-16-16h-32c-8.8 0-16 7.2-16 16v16H16c-8.8 0-16 7.2-16 16v32c0 8.8 7.2 16 16 16h336v16c0 8.8 7.2 16 16 16h32c8.8 0 16-7.2 16-16v-16h80c8.8 0 16-7.2 16-16v-32c0-8.8-7.2-16-16-16zm0-160H288V48c0-8.8-7.2-16-16-16h-32c-8.8 0-16 7.2-16 16v16H16C7.2 64 0 71.2 0 80v32c0 8.8 7.2 16 16 16h208v16c0 8.8 7.2 16 16 16h32c8.8 0 16-7.2 16-16v-16h208c8.8 0 16-7.2 16-16V80c0-8.8-7.2-16-16-16z"
            fill="currentColor"
          />
        </svg>
        <span
          class="text-0-2-5"
        >
          Settings
        </span>
      </button>
    </li>
  </ul>
</DocumentFragment>
`;

exports[`Navigation component matches snapshot 2`] = `
<DocumentFragment>
  <ul
    class="test"
  >
    <li
      class="item-0-2-8"
    >
      <button
        class="button-0-2-9"
        type="button"
      >
        <svg
          aria-hidden="true"
          class="svg-inline--fa fa-caret-square-left fa-w-14 "
          data-icon="caret-square-left"
          data-prefix="far"
          focusable="false"
          role="img"
          viewBox="0 0 448 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M272 157.1v197.8c0 10.7-13 16.1-20.5 8.5l-98.3-98.9c-4.7-4.7-4.7-12.2 0-16.9l98.3-98.9c7.5-7.7 20.5-2.3 20.5 8.4zM448 80v352c0 26.5-21.5 48-48 48H48c-26.5 0-48-21.5-48-48V80c0-26.5 21.5-48 48-48h352c26.5 0 48 21.5 48 48zm-48 346V86c0-3.3-2.7-6-6-6H54c-3.3 0-6 2.7-6 6v340c0 3.3 2.7 6 6 6h340c3.3 0 6-2.7 6-6z"
            fill="currentColor"
          />
        </svg>
        <span
          class="text-0-2-12"
        >
          My offers
        </span>
      </button>
    </li>
    <li
      class="item-0-2-8"
    >
      <button
        class="button-0-2-9"
        type="button"
      >
        <svg
          aria-hidden="true"
          class="svg-inline--fa fa-envelope fa-w-16 "
          data-icon="envelope"
          data-prefix="far"
          focusable="false"
          role="img"
          viewBox="0 0 512 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M464 64H48C21.49 64 0 85.49 0 112v288c0 26.51 21.49 48 48 48h416c26.51 0 48-21.49 48-48V112c0-26.51-21.49-48-48-48zm0 48v40.805c-22.422 18.259-58.168 46.651-134.587 106.49-16.841 13.247-50.201 45.072-73.413 44.701-23.208.375-56.579-31.459-73.413-44.701C106.18 199.465 70.425 171.067 48 152.805V112h416zM48 400V214.398c22.914 18.251 55.409 43.862 104.938 82.646 21.857 17.205 60.134 55.186 103.062 54.955 42.717.231 80.509-37.199 103.053-54.947 49.528-38.783 82.032-64.401 104.947-82.653V400H48z"
            fill="currentColor"
          />
        </svg>
        <span
          class="text-0-2-12"
        >
          My messages
        </span>
        <span
          class="num-0-2-13"
        >
          4
        </span>
        <svg
          aria-hidden="true"
          class="svg-inline--fa fa-chevron-down fa-w-14 "
          data-icon="chevron-down"
          data-prefix="fas"
          focusable="false"
          role="img"
          viewBox="0 0 448 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M207.029 381.476L12.686 187.132c-9.373-9.373-9.373-24.569 0-33.941l22.667-22.667c9.357-9.357 24.522-9.375 33.901-.04L224 284.505l154.745-154.021c9.379-9.335 24.544-9.317 33.901.04l22.667 22.667c9.373 9.373 9.373 24.569 0 33.941L240.971 381.476c-9.373 9.372-24.569 9.372-33.942 0z"
            fill="currentColor"
          />
        </svg>
      </button>
    </li>
    <li
      class="item-0-2-8"
    >
      <button
        class="button-0-2-9"
        type="button"
      >
        <svg
          aria-hidden="true"
          class="svg-inline--fa fa-comment-alt fa-w-16 "
          data-icon="comment-alt"
          data-prefix="far"
          focusable="false"
          role="img"
          viewBox="0 0 512 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M448 0H64C28.7 0 0 28.7 0 64v288c0 35.3 28.7 64 64 64h96v84c0 7.1 5.8 12 12 12 2.4 0 4.9-.7 7.1-2.4L304 416h144c35.3 0 64-28.7 64-64V64c0-35.3-28.7-64-64-64zm16 352c0 8.8-7.2 16-16 16H288l-12.8 9.6L208 428v-60H64c-8.8 0-16-7.2-16-16V64c0-8.8 7.2-16 16-16h384c8.8 0 16 7.2 16 16v288z"
            fill="currentColor"
          />
        </svg>
        <span
          class="text-0-2-12"
        >
          Notifications
        </span>
        <span
          class="num-0-2-13"
        >
          5
        </span>
      </button>
    </li>
    <li
      class="item-0-2-8"
    >
      <button
        class="button-0-2-9"
        type="button"
      >
        <svg
          aria-hidden="true"
          class="svg-inline--fa fa-sliders-h fa-w-16 "
          data-icon="sliders-h"
          data-prefix="fas"
          focusable="false"
          role="img"
          viewBox="0 0 512 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M496 384H160v-16c0-8.8-7.2-16-16-16h-32c-8.8 0-16 7.2-16 16v16H16c-8.8 0-16 7.2-16 16v32c0 8.8 7.2 16 16 16h80v16c0 8.8 7.2 16 16 16h32c8.8 0 16-7.2 16-16v-16h336c8.8 0 16-7.2 16-16v-32c0-8.8-7.2-16-16-16zm0-160h-80v-16c0-8.8-7.2-16-16-16h-32c-8.8 0-16 7.2-16 16v16H16c-8.8 0-16 7.2-16 16v32c0 8.8 7.2 16 16 16h336v16c0 8.8 7.2 16 16 16h32c8.8 0 16-7.2 16-16v-16h80c8.8 0 16-7.2 16-16v-32c0-8.8-7.2-16-16-16zm0-160H288V48c0-8.8-7.2-16-16-16h-32c-8.8 0-16 7.2-16 16v16H16C7.2 64 0 71.2 0 80v32c0 8.8 7.2 16 16 16h208v16c0 8.8 7.2 16 16 16h32c8.8 0 16-7.2 16-16v-16h208c8.8 0 16-7.2 16-16V80c0-8.8-7.2-16-16-16z"
            fill="currentColor"
          />
        </svg>
        <span
          class="text-0-2-12"
        >
          Settings
        </span>
      </button>
    </li>
  </ul>
</DocumentFragment>
`;
